libtransmission/rpc-server.c

```diff
@@ -812,6 +816,30 @@ static void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+static bool bindUnixSocket(struct event_base* base, struct evhttp* httpd, char const* path)
+{
+    unlink(path);
+
+    struct sockaddr_un addr;
+    addr.sun_family = AF_UNIX;
+    strncpy(addr.sun_path, path, sizeof(addr.sun_path) - 1);
+
+    struct evconnlistener* lev = evconnlistener_new_bind(
+        base, NULL, NULL, LEV_OPT_CLOSE_ON_FREE, -1, (struct sockaddr*)&addr, sizeof(addr));
+    if (!lev)
+    {
+        return false;
+    }
+
+    struct evhttp_bound_socket* handle = evhttp_bind_listener(httpd, lev);
+    if (!handle)
+    {
+        return false;
+    }
+
+    return true;
```

Comment:
```suggestion
    return evhttp_bind_listener(httpd, lev) != nullptr;
```

(unless we need to keep the handle somewhere for future cleanup?)